src/OpenTelemetry/Trace/Export/Internal/BroadcastActivityProcessor.cs

```diff
@@ -0,0 +1,105 @@
+﻿// <copyright file="BroadcastActivityProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace.Export.Internal
+{
+    internal class BroadcastActivityProcessor : ActivityProcessor, IDisposable
+    {
+        private readonly IEnumerable<ActivityProcessor> processors;
+
+        public BroadcastActivityProcessor(IEnumerable<ActivityProcessor> processors)
+        {
+            if (processors == null)
+            {
+                throw new ArgumentNullException(nameof(processors));
+            }
+
+            if (!processors.Any())
+            {
+                throw new ArgumentException($"{nameof(processors)} collection is empty");
+            }
+
+            this.processors = processors;
+        }
+
+        public override void OnEnd(Activity activity)
+        {
+            foreach (var processor in this.processors)
+            {
+                try
+                {
+                    processor.OnEnd(activity);
+                }
+                catch (Exception e)
+                {
+                    OpenTelemetrySdkEventSource.Log.SpanProcessorException("OnEnd", e);
+                }
+            }
+        }
+
+        public override void OnStart(Activity activity)
+        {
+            foreach (var processor in this.processors)
+            {
+                try
+                {
+                    processor.OnStart(activity);
+                }
+                catch (Exception e)
+                {
+                    OpenTelemetrySdkEventSource.Log.SpanProcessorException("OnStart", e);
+                }
+            }
+        }
+
+        public override Task ShutdownAsync(CancellationToken cancellationToken)
+        {
+            var tasks = new List<Task>();
+            foreach (var processor in this.processors)
+            {
+                tasks.Add(processor.ShutdownAsync(cancellationToken));
+            }
+
+            return Task.WhenAll(tasks);
+        }
+
+        public void Dispose()
```

Comment:
This is different from what I've normally seen, worth double checking https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implement-the-dispose-pattern.

Reply:
IMO because it's `internal` it's fine as-is. That being said, `OpenTelemetrySdk` isn't calling shutdown or dispose on the `ActivityProcessor` it is maintaining. That's a problem. We need to flush out spans & perform cleanup on app shutdown right? I think OpenTelemetrySdk.Dispose should call ActivityProcessor.Dispose which should invoke ActivityProcessor.ShutdownAsync? We might want to look at using IAsyncDisposable for that.

Reply:
There are few things we need to address:
1. #674 `SimpleSpanProcessor` needs to be a synchronous call (e.g. when the control is returned to the caller, the span got processed - [current it is not guaranteed](https://github.com/open-telemetry/opentelemetry-dotnet/pull/660#discussion_r421891170)).
2. The life-cycle management of processors/exporters, related to this [comment](https://github.com/open-telemetry/opentelemetry-dotnet/pull/660#discussion_r419597541). For example, if we have the same processor/exporter registered twice, would we flush the exporter on the first `OnEnd`, or maintain a refcount and call `OnEnd` during `Dispose`? Does `OnEnd` need to be ref-counted as well?
3. We need to have a deterministic flush semantic. By default I would imagine we want to flush traces/logs before app shutdown, metrics is something debatable. And we need to give option to the user whether they want to wait indefinitely or have a timeout or no wait at all.

Reply:
Agree with all of the above - But my goal with this PR is to just mimic the existing behavior with Spans (this pr was made with pure 'copy paste & minor adjustments' ). I'd prefer to get the parity 1st, then fix underlying issues.

Let me know if this approach is okay?

Reply:
Yeah, these are not blockers for this PR.

Reply:
> IMO because it's `internal` it's fine as-is. That being said, `OpenTelemetrySdk` isn't calling shutdown or dispose on the `ActivityProcessor` it is maintaining. That's a problem. We need to flush out spans & perform cleanup on app shutdown right? I think OpenTelemetrySdk.Dispose should call ActivityProcessor.Dispose which should invoke ActivityProcessor.ShutdownAsync? We might want to look at using IAsyncDisposable for that.

Yes this is needed. Modified OTSDK to call dispose on the Activityprocessor it has.

Reply:
Will follow up on why Span was not calling Shutdown in SimpleProcessor! (Was doing it right for batching one). Anyway - both will be fixed in next PR.